### PR TITLE
Generic/EmptyPHPStatement: improve code coverage

### PR DESCRIPTION
**Pull request by rodrigoprimo**
# Description

This PR improves code coverage for the `Generic.CodeAnalysis.EmptyPHPStatement` sniff.

~While adding tests for the short open tag, I noticed a behavior in the sniff related to #593 that we might want to address in this PR or in a separate PR if it is not intentional behavior. The sniff fixes `<? ; something_else(); ?>` to `<?something_else(); ?>`. Note that there is no space between the PHP open tag and the function call. Should the sniff change to preserve one space between the PHP open tag and the function call? It is probably not a high priority as the resulting code is not a parse error, and short open tags are not very popular.~

_Update: question above moved to https://github.com/PHPCSStandards/PHP_CodeSniffer/issues/728_

## Related issues/external references

Part of https://github.com/PHPCSStandards/PHP_CodeSniffer/issues/146

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [ ] Documentation improvement

## PR checklist

- [x] I have checked there is no other PR open for the same change.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md).
- [x] I grant the project the right to include and distribute the code under the BSD-3-Clause license (and I have the right to grant these rights).
- [x] I have added tests to cover my changes.
- [x] I have verified that the code complies with the projects coding standards.
- [ ] \[Required for new sniffs\] I have added XML documentation for the sniff.

### CODE REVIEW

**Comment by jrfnl** [comment]
Please move the question regarding the behaviour with short open tags to its own issue (linking to this PR). Different decision point, so it doesn't belong in this PR.

File: src/Standards/Generic/Tests/CodeAnalysis/EmptyPHPStatementUnitTest.2.inc
Comment (jrfnl) [inline src/Standards/Generic/Tests/CodeAnalysis/EmptyPHPStatementUnitTest.2.inc:1]: Copy/pasta ? These aren't short open echo tags....

File: src/Standards/Generic/Sniffs/CodeAnalysis/EmptyPHPStatementSniff.php
Comment (jrfnl) [inline src/Standards/Generic/Sniffs/CodeAnalysis/EmptyPHPStatementSniff.php:56]: Could the same condition on line 127 be removed as well ?

**Review by jrfnl — COMMENTED** [review]
Looking good. Left a few nitpicks in-line, but other than that, all good.

Now wondering whether the sniff would benefit from a test with multiple superfluous semi-colons after a statement and/or between tags ?

**Comment by rodrigoprimo** [comment]
> Please move the question regarding the behaviour with short open tags to its own issue (linking to this PR). Different decision point, so it doesn't belong in this PR.

https://github.com/PHPCSStandards/PHP_CodeSniffer/issues/728

File: src/Standards/Generic/Tests/CodeAnalysis/EmptyPHPStatementUnitTest.2.inc
Comment (rodrigoprimo) [reply src/Standards/Generic/Tests/CodeAnalysis/EmptyPHPStatementUnitTest.2.inc]: Good catch. Fixed.

File: src/Standards/Generic/Sniffs/CodeAnalysis/EmptyPHPStatementSniff.php
Comment (rodrigoprimo) [reply src/Standards/Generic/Sniffs/CodeAnalysis/EmptyPHPStatementSniff.php:56]: I believe so, just removed it in a new commit.

**Comment by rodrigoprimo** [comment]
Thanks for reviewing this PR, @jrfnl! I implemented everything you suggested, and this is now ready for a second review.

> Now wondering whether the sniff would benefit from a test with multiple superfluous semi-colons after a statement and/or between tags ?

I think it is a good idea. There is already a test with two superfluous semi-colons after a class definition, so I updated other existing tests so that we also have a test with two superfluous semi-colons between PHP tags and a test with two superfluous semi-colons between PHP tags and after a statement.

**Comment by rodrigoprimo** [comment]
@jrfnl, just FYI, the Windows build failed, and I don't think the problem is related to the changes in the PR: https://github.com/PHPCSStandards/PHP_CodeSniffer/actions/runs/12016963721/job/33498304890?pr=699

It seems it was a Coveralls hiccup: https://github.com/PHPCSStandards/PHP_CodeSniffer/actions/runs/12016963721/job/33498304890?pr=699#step:19:193

**Comment by rodrigoprimo** [comment]
I believe you ran the build again while I was typing the previous comment and it passed. Thanks!

**Review by jrfnl — APPROVED** [review]
Thanks @rodrigoprimo! I'll rebase & squash select commits and will merge this PR once the build has passed.

**Pull request merged**
